packages/data/src/hypoCompress.ts

```diff
@@ -0,0 +1,406 @@
+import { isEqual } from "lodash";
+import { HYPO_PLAYER_NAMES, MAX_PLAYERS, MIN_PLAYERS } from "./constants";
+import { getVariant, getVariantByID } from "./gameData";
+import { parseIntSafe } from "./utils";
+
+interface DeckCard {
+  suitIndex: number;
+  rank: number;
+}
+
+interface Action {
+  type: number;
+  target: number;
+  value?: number;
+}
+
+interface Options {
+  variant: string | undefined;
+}
+
+interface MinMax {
+  min: number;
+  max: number;
+}
+
+export interface GameJSON {
+  players: string[];
+  deck: DeckCard[];
+  actions: Action[];
+  options: Options;
+  characters: string[];
+  id: number;
+  notes: [];
+  seed: string;
+}
+
+const BASE62 = "abcdefghijklmnopqrstuvwxyz0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+
+/**
+ * Compresses a string representing a GameJSON object.
+ * Returns null if the compression fails.
+ *
+ * The resulting string is composed of three substrings separated by comma.
+ * The first substring represents the number of players and the deck.
+ * The second substring represents the actions.
+ * The third substring is the ID of the variant.
+ *
+ * Finally hyphens are added to the string to make URL text wrap when posting.
+ *
+ * @param JSONString A string representing a hypo JSON
```

Comment:
jsodc has to have complete sentences

Reply:
?

Reply:
its missing a period at the end, also use "hypothetical" instead of "hypo" to avoid confusion.

Reply:
(see https://jsdoc.app/tags-example.html)